### PR TITLE
Remove `ffi_returns_twice` feature

### PR DESCRIPTION
**Pull request by clubby789**
The [tracking issue](https://github.com/rust-lang/rust/issues/58314) and [RFC](https://github.com/rust-lang/rfcs/pull/2633) have been closed for a couple of years.

There is also an attribute gate in R-A which should be removed if this lands.

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/llvm/ffi.rs [hunk 1]
```diff
@@ -185,7 +185,6 @@ pub enum AttributeKind {
     SanitizeMemory = 22,
     NonLazyBind = 23,
     OptimizeNone = 24,
-    ReturnsTwice = 25,
```
Comment (clubby789) [inline compiler/rustc_codegen_llvm/src/llvm/ffi.rs:188]: Didn't renumber the following variants as `27` is also missing

File: compiler/rustc_error_codes/src/error_codes/E0724.md [hunk 2]
```diff
@@ -1,9 +1,11 @@
+#### Note: this error code is no longer emitted by the compiler.
+
 `#[ffi_returns_twice]` was used on something other than a foreign function
```
Comment (clubby789) [inline compiler/rustc_error_codes/src/error_codes/E0724.md:3]: Not sure if this error code should be removed completely?

File: compiler/rustc_error_codes/src/error_codes/E0724.md [hunk 3]
```diff
@@ -1,25 +0,0 @@
-`#[ffi_returns_twice]` was used on something other than a foreign function
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/error_codes/E0724.md:1]: Please revert this change. Error code explanations should never be removed, just updated like you can see in https://doc.rust-lang.org/error_codes/E0729.html (CI will not pass until then in any case).

File: compiler/rustc_error_codes/src/lib.rs [hunk 4]
```diff
@@ -675,5 +674,6 @@ E0797: 0797,
 //  E0709, // multiple different lifetimes used in arguments of `async fn`
 //  E0721, // `await` keyword
 //  E0723, // unstable feature in `const` context
+//  E0724, // `ffi_returns_twice` on something other than foreign function
```
Comment (GuillaumeGomez) [inline compiler/rustc_error_codes/src/lib.rs:677]: Please revert this change.

**Comment by BatmanAoD** [comment]
I think it's correct to remove this. Removal was previously alluded to [in our zulip](https://rust-lang.zulipchat.com/#narrow/stream/210922-project-ffi-unwind/topic/cost.20of.20supporting.20longjmp.20without.20annotations/near/383342346), and I think @Amanieu speaks for the whole of the working group when [describing returns-twice](https://rust-lang.zulipchat.com/#narrow/stream/210922-project-ffi-unwind/topic/cost.20of.20supporting.20longjmp.20without.20annotations/near/375298042) as a "horrible" hack "even by C standards."
